tests/components/homekit_controller/specific_devices/test_aqara_gateway.py

```diff
@@ -0,0 +1,45 @@
+"""
+Regression tests for Aqara Gateway V3.
+
+https://github.com/home-assistant/home-assistant/issues/20957
+"""
+
+import os
+
+from homeassistant.components.light import SUPPORT_BRIGHTNESS, SUPPORT_COLOR
+from tests.components.homekit_controller.common import (
+    setup_accessories_from_file, setup_test_accessories, Helper
+)
+
+
+async def test_aqara_gateway_setup(hass):
+    """Test that a Aqara Gateway can be correctly setup in HA."""
+    profile_path = os.path.join(
+        os.path.dirname(__file__), 'aqara_gateway.json'
+    )
+    accessories = setup_accessories_from_file(profile_path)
```

Comment:
This does I/O and should be put on the executor. We shouldn't interfere with the event loop even if it's a test, I think.

```py
await hass.async_add_executor_job(...)
```